src/databricks/labs/ucx/workspace_access/generic.py

```diff
@@ -208,38 +217,66 @@ def _prepare_new_acl(
         return acl_requests
 
 
-class WorkspaceListing(Listing):
-    def __init__(self, ws: WorkspaceClient, num_threads=20, start_path: str | None = "/"):
-        super().__init__(..., ..., ...)
+class WorkspaceListing(Listing, CrawlerBase):
+    def __init__(
+        self,
+        ws: WorkspaceClient,
+        sql_backend: SqlBackend,
+        inventory_database: str,
+        num_threads=20,
+        start_path: str | None = "/",
+    ):
+        Listing.__init__(self, ..., ..., ...)
+        CrawlerBase.__init__(
+            self,
+            backend=sql_backend,
+            catalog="hive_metastore",
+            schema=inventory_database,
+            table="workspace_objects",
+            klass=WorkspaceObjectInfo,
+        )
         self._ws = ws
         self._num_threads = num_threads
         self._start_path = start_path
+        self._sql_backend = sql_backend
```

Comment:
You don't have to assign these two fields, it's already done by crawlerbase

Reply:
good point

Reply:
will update in the next issue PR